extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -1012,15 +1014,15 @@ private void possiblyResetOffsetsOrWait(
         // and the current message offset in the kafka partition is more than the
         // next message offset that we are trying to fetch
         if (leastAvailableOffset > nextOffset) {
```

Comment:
There is this comment in the code before consumer polls data -
```
// The retrying business is because the KafkaConsumer throws OffsetOutOfRangeException if the seeked-to
// offset is not present in the topic-partition. This can happen if we're asking a task to read from data
// that has not been written yet (which is totally legitimate). So let's wait for it to show up.
```
This suggests that it is ok to wait for data to show up in case (leastAvailableOffset < nextOffset) and we should not reset. However, it is definitely different from the case where topic was deleted and re-created in which case we should reset. So not sure what should be done. 

May be resetting in all cases is not bad, however if new data does not show up we will keep on resetting but seems like a rare scenario. What do you think ?